ledger/acctupdates.go

```diff
@@ -2138,26 +2152,53 @@ func (au *accountUpdates) commitRound(offset uint64, dbRound basics.Round, lookb
 
 		db.ResetTransactionWarnDeadline(ctx, tx, time.Now().Add(accountsUpdatePerRoundHighWatermark*time.Duration(offset)))
 
-		err = compactDeltas.accountsLoadOld(tx)
-		if err != nil {
-			return err
+		if updateStats {
+			begin := time.Now()
+			err = compactDeltas.accountsLoadOld(tx)
+			if err != nil {
+				return err
+			}
+			stats.OldAccountPreloadDuration = time.Now().Sub(begin)
+		} else {
+			err = compactDeltas.accountsLoadOld(tx)
+			if err != nil {
+				return err
+			}
```

Comment:
this isn't incorrect, but I think that duplicating the "actual work code" is not desired and would create more confusion.
what do you think about the following pattern: 
```go
if updateStats {
      stats.OldAccountPreloadDuration = time.Now().UnixNano()
}
err = compactDeltas.accountsLoadOld(tx)
if err != nil {
	return err
}
if updateStats {
      stats.OldAccountPreloadDuration = time.Now().UnixNano() - stats.OldAccountPreloadDuration
}
```

Note that in case of a back to back timing, you'd like to have only a single `time.Now()` call.

Reply:
I thought having fewer `if updateStats` checks would keep the code more readable. But this pattern actually makes more sense since duplicating the actual work code would require duplicating any future changes, too.